lib/models/ObjectMDArchive.js

```diff
@@ -0,0 +1,171 @@
+                data.restoreRequestedDays,
+                data.restoreCompletedAt,
+                data.restoreWillExpireAt,
+            );
+            return !!object;
+        } catch (err) {
+            return false;
+        }
+    }
+
+    /**
+     *
+     * @returns {Object} archiveInfo
+     */
+    getArchiveInfo() {
+        return this.archiveInfo;
+    }
+
+    /**
+     * @param {Object} value archiveInfo
+     * @returns {void}
+     * @throws {Error} case of invalid parameter
+     */
+    setArchiveInfo(value) {
+        if (!value) {
+            throw new Error('archiveInfo is required.');
+        } else if (typeof value !== 'object') {
+            throw new Error('archiveInfo must be type of object.');
+        }
+        this.archiveInfo = value;
+    }
+
+    /**
+     *
+     * @returns {Date} restoreRequestedAt
+     */
+    getRestoreRequestedAt() {
+        return this.restoreRequestedAt;
+    }
+    /**
+     * @param {Object} value restoreRequestedAt
+     * @returns {void}
+     * @throws {Error} case of invalid parameter
+     */
+    setRestoreRequestedAt(value) {
+        if (value) {
+            if (!(value instanceof Date)) {
+                throw new Error('restoreRequestedAt must be type of Date.');
+            }
+            this.restoreRequestedAt = value;
+        }
+    }
+
+    /**
+     *
+     * @returns {Number} restoreRequestedDays
+     */
+    getRestoreRequestedDays() {
+        return this.restoreRequestedDays;
+    }
+    /**
+     * @param {Number} value restoreRequestedDays
+     * @returns {void}
+     * @throws {Error} case of invalid parameter
+     */
+    setRestoreRequestedDays(value) {
+        if (value) {
+            if (isNaN(value)) {
+                throw new Error('restoreRequestedDays must be type of Number.');
+            }
+            this.restoreRequestedDays = value;
+        }
+    }
+
+    /**
+     *
+     * @returns {Date} restoreCompletedAt
+     */
+    getRestoreCompletedAt() {
+        return this.restoreCompletedAt;
+    }
+    /**
+     * @param {Date} value restoreCompletedAt
+     * @returns {void}
+     * @throws {Error} case of invalid parameter
+     */
+    setRestoreCompletedAt(value) {
+        if (value) {
+            if (!this.restoreRequestedAt || !this.restoreRequestedDays) {
+                throw new Error('restoreCompletedAt must be set after restoreRequestedAt and restoreRequestedDays.');
+            }
+            if (!(value instanceof Date)) {
+                throw new Error('restoreCompletedAt must be type of Date.');
+            }
+            this.restoreCompletedAt = value;
+        }
+    }
+    /**
+     *
+     * @returns {Date} restoreWillExpireAt
+     */
+    getRestoreWillExpireAt() {
+        return this.restoreWillExpireAt;
+    }
+    /**
+     * @param {Date} value restoreWillExpireAt
+     * @returns {void}
+     * @throws {Error} case of invalid parameter
+     */
+    setRestoreWillExpireAt(value) {
```

Comment:
I'm supposing there will be a function like 'handleRestore' in the future that will calculate the value seperately then call this?

Reply:
It will be done in the APIs, with internal logic :slightly_smiling_face: 
But this object will mainly be edited by backbeat and the TLP client (the "driver" we will use to communicate with a TAPE cold backend), in an async way, so here we only need to provide a class with a set of getter/setters to let the service properly manage the metadata.